test/tests/functional/pbs_peer.py

```diff
@@ -0,0 +1,147 @@
+        if not name:
+            name = self.cres
+
+        if not server:
+            server = self.server
+
+        server.manager(MGR_CMD_CREATE, RSC,
+                       {'type': 'long', 'flag': 'q'},
+                       id=name)
+
+        self.scheduler.add_resource(name)
+
+    def create_queue(self, name=None, server=None, a=None):
+        if not name:
+            name = self.cqueue
+
+        if not server:
+            server = self.server
+
+        if not a:
+            a = {'queue_type': 'execution', 'enabled': True, 'started': True}
+
+        server.manager(MGR_CMD_CREATE, QUEUE, a, id=name)
+
+    def test_local_resc_limits(self):
+        """
+        Test that a local peering queue enforces new limits
+        """
+        self.create_resource()
+        self.create_queue()
+        a = {'resources_max.' + self.cres: 4}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id=self.cqueue)
+        a = {'started': False}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id='workq')
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id=self.cqueue)
+
+        a = {'peer_queue': '"' + self.cqueue + ' workq"'}
+        self.scheduler.set_sched_config(a)
+
+        j = Job(TEST_USER, attrs={'Resource_List.' + self.cres: 100})
+        jid = self.server.submit(j)
+        j = Job(TEST_USER, attrs={'Resource_List.' + self.cres: 1})
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        a = {'started': True}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id=self.cqueue)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': True})
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+        msg = (jid + ';Failed to run: Job violates queue and/or server'
+               ' resource limits (15036)')
+        self.scheduler.log_match(msg)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+
+    @requirements(num_servers=2)
+    def test_remote_resc_limits(self):
+        """
+        Test that a remote peering queue enforces new limits
+        """
+        s1 = self.servers.values()[0]
+        s2 = self.servers.values()[1]
+        self.create_resource(server=s1)
+        self.create_resource(server=s2)
+        a = {'resources_max.' + self.cres: 4}
+        s1.manager(MGR_CMD_SET, QUEUE, a, id='workq')
+        a = {'started': False}
+        s1.manager(MGR_CMD_SET, QUEUE, a, id='workq')
+        s2.manager(MGR_CMD_SET, QUEUE, a, id='workq')
+
+        a = {'flatuid': True}
+        s1.manager(MGR_CMD_SET, SERVER, a)
+        s2.manager(MGR_CMD_SET, SERVER, a)
+
+        a = {'peer_queue': '"workq workq@' + s2.hostname + '"'}
+        self.scheduler.set_sched_config(a)
+
+        a = {'Resource_List.' + self.cres: 100,
+             ATTR_queue: 'workq@' + s2.hostname}
+        j = Job(TEST_USER, attrs=a)
+        jid = s1.submit(j)
```

Comment:
If s1 is the pulling complex then shouldn't this be s2.submit()?

Reply:
PTL is a little screwy and won't let me submit from s2. However, because I set the queue to workq@s2, it will submit the job from s1's host to s2.

Reply:
To be more clear, the first job submits just fine, the second fails with "Failed to receive data from background qsub" or something like that.